### PR TITLE
Unsaved group resolve

### PR DESCRIPTION
**Pull request by yarovyii**
PQSWPRG-8012

### CODE REVIEW

File: rest/conf/54_agroups.xml.in [hunk 1]
```diff
@@ -7,6 +7,15 @@
     </args>
 </mapping>
 
+<mapping>
+    <target>agroup/resolve@lib${NAME}-rest</target>
+    <url>^/api/v1/automaticgroup/(.+)/posti$</url>
```
Comment (perrettecl) [inline rest/conf/54_agroups.xml.in:12]: not sure it's the best url for the resolve without id

**Review by perrettecl — COMMENTED** [review]
Please make sure the id 0 is not a valid id and also rework the restAPI handler because seems to be in the wrong place, maybe better to have it in /api/v1/automaticgroup/content for the resolve without id

File: rest/src/resolve.cpp [hunk 2]
```diff
@@ -2,9 +2,10 @@
 #include "common/commands.h"
 #include "common/message-bus.h"
 #include "group-rest.h"
-#include <fty/rest/component.h>
 #include <asset/json.h>
+#include <fty/rest/component.h>
 #include <fty/split.h>
+#include <fty_log.h>
```
Comment (etnKlendathu) [inline rest/src/resolve.cpp:8]: why?

File: rest/src/resolve.cpp [hunk 3]
```diff
@@ -28,7 +29,19 @@ unsigned Resolve::run()
...
-    in.id = fty::convert<uint16_t>(*strIdPrt);
+    if (strIdPrt.value() != "0") {
+        in.id = fty::convert<uint16_t>(*strIdPrt);
+    } else {
+        std::string jsonBody = m_request.body();
+        if (jsonBody.empty()) {
+            throw rest::errors::BadInput("Group payload is empty");
+        }
+
+        in.id = 0;
```
Comment (etnKlendathu) [inline rest/src/resolve.cpp:40]: reason to set id here?

**Review by etnKlendathu — COMMENTED** [review]
looks almost good

File: rest/src/resolve.cpp [hunk 4]
```diff
@@ -16,9 +15,14 @@ unsigned Resolve::run()
         throw rest::Error(ret.error());
     }
 
-    auto strIdPrt = m_request.queryArg<std::string>("id");
+    auto        strIdPrt = m_request.queryArg<std::string>("id");
```
Comment (etnKlendathu) [inline rest/src/resolve.cpp:18]: you removed id from the url definition... why are you trying to get it in code? This id will never exists :)

Comment (yarovyii) [reply rest/src/resolve.cpp:18]: we have two mapping and one of them(GET) with id

Comment (etnKlendathu) [reply rest/src/resolve.cpp:18]: I'm not sure if it work at all... possible you can't map different urls to 1 callback

**Pull request merged**